invenio_preservation_sync/resources/resource.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Invenio Preservation Sync module to create REST APIs."""
+
+from flask import g
+from flask_resources import (
+    Resource,
+    from_conf,
+    request_parser,
+    resource_requestctx,
+    response_handler,
+    route,
+)
+
+from ..errors import ErrorHandlersMixin
+from ..proxies import current_preservation_sync_service as service
+
+request_view_args = request_parser(from_conf("request_view_args"), location="view_args")
+
+
+class PreservationInfoResource(ErrorHandlersMixin, Resource):
+    """Preservation Info resource."""
+
+    def __init__(self, config, latest_path=None, list_path=None):
+        """Constructor."""
+        super().__init__(config)
+        self.latest_route = latest_path
+        self.list_route = list_path
+
+    def create_url_rules(self):
+        """Create the URL rules for the preservation info resource.
+
+        * **GET** latest preservation path can be configured in :attr:`invenio_preservation_sync.config.PRESERVATION_SYNC_GET_LATEST_PATH`.
+            (Default: */records/<pid_id>/preservations/latest*).
+
+        * **GET** list of preservations path can be configured in :attr:`invenio_preservation_sync.config.PRESERVATION_SYNC_GET_LIST_PATH`
+             (Default: */records/<pid_id>/preservations/*).
+        """
+        routes = self.config.routes
+        if not self.latest_route:
+            self.latest_route = routes["latest"]
+        if not self.list_route:
+            self.list_route = routes["list"]
+        return [
+            route("GET", self.latest_route, self.get_latest),
+            route("GET", self.list_route, self.get_list),
+        ]
+
+    @request_view_args
+    @response_handler()
+    def get_latest(self):
+        """GET endpoint to return the latest preservation info for a given record.
+
+        Request param: **pid_id** PersistentIdentifier ID for the record.
+
+        :returns: Response status code, see *message* for more details
+
+        * **200** - Latest preservation is returned as a JSON object.
+        * **400** - Parameter was not valid.
+        * **403** - Permission requirement was not met.
+        * **404** - PID could not be resolved or there are no preservations for the given record or the module is disabled.
+        * **503** - Mandatory config was missing.
+        """
+        pid_id = resource_requestctx.view_args["pid_id"]
+        preservation = service.read(g.identity, pid_id, latest=True)
```

Comment:
The service should be injected in the constructor (dependency injection, Java wins here :D), and not using a global import:
```suggestion
        preservation = self.service.read(g.identity, pid_id, latest=True)
```
See [an example here](https://github.com/inveniosoftware/invenio-jobs/blob/master/invenio_jobs/resources/resources.py#L95). It is injected [here](https://github.com/inveniosoftware/invenio-jobs/blob/master/invenio_jobs/ext.py#L72).